include/dmlc/optional.h

```diff
@@ -107,6 +108,8 @@ class optional {
   }
   /*! \brief whether this object is holding a value */
   explicit operator bool() const { return !is_none; }
+  /*! \brief alternate, more-explicit check of object value */
+  const bool IsNone() const { return is_none; }
```

Comment:
I removed the pointless initial 'const' keyword- there by mistake.  Is this what you were referring to?